cpp/core/shuffle/rss/RssPartitionWriter.cc

```diff
@@ -85,7 +85,7 @@ arrow::Status RssPartitionWriter::doEvict(
       inMemoryPayload->toBlockPayload(
           payloadType, payloadPool_.get(), codec_ ? codec_.get() : nullptr, std::move(compressed)));
   // Copy payload to arrow buffered os.
-  ARROW_ASSIGN_OR_RAISE(auto rssBufferOs, arrow::io::BufferOutputStream::Create(options_.pushBufferMaxSize, pool_));
+  ARROW_ASSIGN_OR_RAISE(auto rssBufferOs, arrow::io::BufferOutputStream::Create(options_.pushBufferMaxSize));
```

Comment:
Thanks for working on a fix. Though this could turn the memory buffer into untracked? 

cc @marin-ma

Reply:
It is, actually I'm a little confused about delegate the timing of spill trigger to spark framework, which means we can't make any memory allocation when spilling.

Reply:
Maybe we should have a threshold to indicate whether trigger spill actively or not?